registrations/tests/test_send_message.py

```diff
@@ -60,77 +63,114 @@ def test_email_is_sent_to_responsible_signups(api_client, registration, user):
-    send_message_data = {"subject": "Message subject", "body": "Message body"}
+    send_message_data = {
+        "subject": "Message subject",
+        "body": "Message body",
+        "signup_groups": [signup_group1.pk],
+    }
 
     assert_send_message(
-        api_client,
-        registration.id,
-        send_message_data,
-        [first_signup.email, second_signup.email, third_signup.email],
+        api_client, registration.id, send_message_data, [third_signup.email]
     )
     # Default language for the email is Finnish
     assert "Tarkastele ilmoittautumistasi täällä" in str(mail.outbox[0].alternatives[0])
 
 
 @pytest.mark.django_db
-def test_email_is_sent_to_mixed_signups(api_client, registration, user):
-    signup_group = SignUpGroupFactory(registration=registration)
-    first_signup = SignUpFactory(
-        signup_group=signup_group,
+def test_email_is_sent_to_selected_signups_only(api_client, registration, user):
+    signup_group0 = SignUpGroupFactory(registration=registration)
+    SignUpFactory(
+        signup_group=signup_group0, registration=registration, email="test@test.com"
+    )
+    responsible_signup = SignUpFactory(
+        signup_group=signup_group0,
         registration=registration,
-        email="test@test.com",
         responsible_for_group=True,
+        email="test2@test.com",
     )
-    SignUpFactory(
-        signup_group=signup_group, registration=registration, email="test2@test.com"
-    )
-    third_signup = SignUpFactory(registration=registration, email="test3@test.com")
 
-    second_signup_group = SignUpGroupFactory(registration=registration)
-    fourth_signup = SignUpFactory(
-        signup_group=second_signup_group,
+    signup_group1 = SignUpGroupFactory(registration=registration)
+    SignUpFactory(
+        signup_group=signup_group1,
         registration=registration,
-        email="test4@test.com",
+        responsible_for_group=True,
+        email="test3@test.com",
     )
+    non_responsible_signup = SignUpFactory(
+        signup_group=signup_group1, registration=registration, email="test4@test.com"
+    )
+
+    individual_signup = SignUpFactory(registration=registration, email="test5@test.com")
+    SignUpFactory(registration=registration, email="test6@test.com")
 
     api_client.force_authenticate(user)
-    send_message_data = {"subject": "Message subject", "body": "Message body"}
+    send_message_data = {
+        "subject": "Message subject",
+        "body": "Message body",
+        "signups": [
+            responsible_signup.pk,
+            non_responsible_signup.pk,
+            individual_signup.pk,
+        ],
+    }
 
     assert_send_message(
         api_client,
         registration.id,
         send_message_data,
-        [first_signup.email, third_signup.email, fourth_signup.email],
+        [
+            responsible_signup.email,
+            non_responsible_signup.email,
+            individual_signup.email,
+        ],
```

Comment:
The test is named `test_email_is_sent_to_selected_signups_only`; however, there seems to be a lot more going than that (multiple signup groups, responsible/non-responsible/individual signups). What's happening here? Is the name correct?

Reply:
The test name is correct because when giving `signups` in the request body, only those should get the emails. There are multiple other ones here to test that they don't also get the email.